### DISCUSSION TITLE
Test results

### DISCUSSION THREAD

**Discussion by wonder99** [springbok/MLM2PRO-GSPro-Connector#23] [General]
A big update indeed!
Here's what I found.  I chose Android from the first menu, and fixed up the window name.  I see the ROI order has changed, so my first attempt at setting the boxes was wrong.  I corrected that, and it reported shots correctly.  I hit ctrl-c to exit the connector and got

```
CONNECTOR || Shutting down connector...
Traceback (most recent call last):
  File "C:\BitBucket\MLM2PRO-GSPro-Connector\cli.py", line 4, in <module>
    main()
  File "C:\BitBucket\MLM2PRO-GSPro-Connector\src\main.py", line 85, in main
    app.process_manager.run()
  File "C:\BitBucket\MLM2PRO-GSPro-Connector\src\process_manager.py", line 35, in run
    if not self.processes_paused and datetime.now() > self.scheduled_time:
KeyboardInterrupt
^C

```

I relaunched the connector and chose Android.  It was unable to process any more shots, with these messages:
```

CONNECTOR || Process shot_process: Error: Could not convert value 0 for 'Ball Speed' to float 0
CONNECTOR || Process shot_process: Error: Could not convert value 0 for 'Ball Speed' to float 0
CONNECTOR || Process shot_process: Error: Could not convert value 0 for 'Ball Speed' to float 0
Exception in thread Thread-4:
```

**Comment by springbok** [comment]
Thanks for testing, 

I must have changed the order of the rio's when redoing the code, I'll see if I can restore it to the original order.

I don't have an android device at the moment so can't test it using android, I've seen those errors when the screen mirror app is minimized, could that have been the case or was the window visible? 

I added code to check if the window is minimized and to restore it if so, but haven't got it to work as yet, will have another look today.

I'll also add code to cleanly exit on ctrl-c

**Comment by springbok** [comment]
Hopefully addressed the issues you found:

1. Cleanly handle ctrl+c
2. Use the same ROI order as the other connector
3. If the mirror app window was minimized it caused errors, added code to check if the window is minimized if so it will restore it

I've created the first beta release which contains the executable and tessoract file you can find it [here](https://github.com/springbok/MLM2PRO-GSPro-Connector/releases/tag/V0.1). Just extract to a directory of your choosing and run it. Don't think you need to run it as Administrator as it worked fine for me as normal user.
